### ISSUE TITLE
VS version selected does not contain required workloads

### ISSUE THREAD

**Issue by maheshsabnis** [dotnet/upgrade-assistant#723]
I am trying to upgrade ASP.NET MVC Project targeted to .NET Framework 4.8 to .NET 5 using the .NET Upgrade Assistant  tool. I am using VS2019 with update 16.10.3.  I have followed steps from "https://docs.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-aspnetmvc" but I am getting the following error:

- Microsoft .NET Upgrade Assistant v0.2.236301+ab7048ed405e48f681c02ce16f691fc473a2d96a -
-----------------------------------------------------------------------------------------

[18:19:12 INF] Loaded 3 extensions
[18:19:13 INF] MSBuild registered from C:\Program Files\dotnet\sdk\5.0.302\
[18:19:15 INF] Initializing upgrade step Select an entrypoint
[18:19:15 INF] Setting entrypoint to only project in solution: C:\DXC\july21\MUV5_NewApp\MUV5_NewApp\MUV5_NewApp.csproj
[18:19:15 INF] Initializing upgrade step Select project to upgrade
[18:19:15 ERR] Project C:\DXC\july21\MUV5_NewApp\MUV5_NewApp\MUV5_NewApp.csproj cannot be loaded: The imported project "C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Microsoft\VisualStudio\v16.0\WebApplications\Microsoft.WebApplication.targets" was not found. Confirm that the expression in the Import declaration "C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Microsoft\VisualStudio\v16.0\WebApplications\Microsoft.WebApplication.targets" is correct, and that the file exists on disk.  C:\DXC\july21\MUV5_NewApp\MUV5_NewApp\MUV5_NewApp.csproj
[18:19:15 INF] No projects need upgraded for selected entrypoint
[18:19:15 INF] Initializing upgrade step Finalize upgrade

Upgrade Steps

I am seeking help on this.
Thanks in advance
Regards
Mahesh Sabnis

**Comment by twsouthwick** [comment]
Looks like you don't have the web application workload in VS 2022.

Long term, we should correlate VS selection better with the version of .NET Core

**Comment by twsouthwick** [comment]
I've opened a design proposal that we'll keep open in lieu of this issue. Please reopen if the workaround does not work for you

**Comment by KSchlobohm** [comment]
@twsouthwick 
I don't think this is a selected workloads issue. The path itself appears to be incorrect. The correct path for the templates should either include `v17.0` or `2019`. I ran into the same issue when I installed Dev17 and I have the Web application workload installed.

One way to hack around this issue is to copy the `v16.0` templates into the path where the application is looking for these files.
